test/Libraries/DynamoMSOfficeTests/ExcelTests.cs

```diff
@@ -983,6 +979,247 @@ public void TestWriteFunctionObjectToExcel()
+            
+            var list = watch.CachedValue.GetElements();
+
+            // input array is empty list, {}
+            // Ensure output is also empty list
+            Assert.AreEqual(0, list.Count);
+
+            Assert.IsTrue(watch.CachedValue.IsCollection);
+            Assert.IsNull(watch.CachedValue.Data);
+   
+           
+        }
+       [Test]
+        public void WriteToReadOnlyFile()
+        {
+
+            // asks for overwriting of file- need to find a way to ignore that.
+
+            string openPath = Path.Combine(TestDirectory, @"core\excel\WriteToReadOnlyFile.dyn");
+            ViewModel.OpenCommand.Execute(openPath);
+
+            var filename = ViewModel.Model.CurrentWorkspace.FirstNodeFromWorkspace<DSCore.File.Filename>();
+
+            // remap the filename as Excel requires an absolute path
+            filename.Value = filename.Value.Replace(@"..\..\..\test", TestDirectory);
+
+            var watch = ViewModel.Model.CurrentWorkspace.GetDSFunctionNodeFromWorkspace("Excel.WriteToFile");
+
+            ViewModel.HomeSpace.Run();
+            Assert.IsNull(watch.CachedValue.Data);
+            Assert.IsTrue(watch.State == Models.ElementState.Warning);
+           
+
+        }
+        
+        [Test]
+        public void WriteModifyFilename()
+        {
+
+            // 1.  Write into  an Existing excel file and Run
+            // 2.  Modify the file name in file path to an existing file
+            //3.  Rerun the file  
+            string openPath = Path.Combine(TestDirectory, @"core\excel\WriteModifyPath.dyn");
+            ViewModel.OpenCommand.Execute(openPath);
+
+            var filename = ViewModel.Model.CurrentWorkspace.FirstNodeFromWorkspace<DSCore.File.Filename>();
+
+            // remap the filename as Excel requires an absolute path
+            filename.Value = filename.Value.Replace(@"..\..\..\test", TestDirectory);
+
+
+            ViewModel.HomeSpace.Run();
+
+            var watch = ViewModel.Model.CurrentWorkspace.GetDSFunctionNodeFromWorkspace("Excel.WriteToFile");
+            var watch2 = ViewModel.Model.CurrentWorkspace.NodeFromWorkspace("5e6a2a5b-7c69-4ab6-b33d-f27267fb2762");
+             Assert.IsTrue(watch.CachedValue.IsCollection);
+            var list1 = watch.CachedValue.GetElements();
+            Assert.AreEqual(6, list1.Count());
+            Assert.IsTrue(list1[0].IsCollection);
+            var rowList = list1[0].GetElements();
+            Assert.AreEqual(5, rowList[0].Data);
+
+            // change the file path 
+
+            ConnectorModel.Make(watch2, watch, 0, 0);
+            ViewModel.HomeSpace.Run();
+
+            Assert.IsTrue(watch.CachedValue.IsCollection);
+            var data = new object[] { new object[] { 5 }, new object[] { 6 }, new object[] { 7 }, new object[] { 8 }, new object[] { 9 }, new object[] { 10 } };
+            AssertPreviewValue(watch.GUID.ToString(), data);
+
+
+        }
+        [Test]
+        public void WriteModifyGraph()
```

Comment:
This test case should pass when [this fix] (https://github.com/DynamoDS/Dynamo/pull/4615) is merged